### PR TITLE
fix: ast for constants

### LINKED ISSUE #210: [bug] invalid ast for `const`
Input:
```php
const CONSTANT = 'Hello', TEST = 'world';
```

Output each `const` as separate item, but should have same ast as for `static`. I will look how we can fix it.

### PR DESCRIPTION
**Pull request by evilebottnawi**
fixes #210

- Introduce `ConstantStatement` node
- `constant` node now have only `name` and `value`
- `ClassConstant` extends `ConstantStatement` and parse flags

### CODE REVIEW

File: src/ast/constantstatement.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+const KIND = "constantstatement";
+
+/**
+ * Declares a static variable into the current scope
+ * @constructor Static
+ * @extends {Statement}
+ * @property {Variable[]|Assign[]} items
+ */
+module.exports = Statement.extends(KIND, function ConstantStatement(
+  kind,
```
Comment (ichiriac) [inline src/ast/constantstatement.js]: here it means you either pass the kind from a parent class (inheritance), or null it you want to use current node kind (constantstatement).

File: src/parser/statement.js [hunk 2]
```diff
@@ -51,8 +51,12 @@ module.exports = {
         return this.read_trait();
       case this.tok.T_USE:
         return this.read_use_statement();
-      case this.tok.T_CONST:
-        return this.next().read_const_list();
+      case this.tok.T_CONST: {
+        const result = this.node("constantstatement");
+        const items = this.next().read_const_list();
+        this.expectEndOfStatement();
+        return result(items);
```
Comment (ichiriac) [inline src/parser/statement.js]: here you need on the first parameter to pass null (related to the kind parameter)

File: src/ast/constantstatement.js (hunk 1 above)
Comment (evilebottnawi) [reply src/ast/constantstatement.js]: @ichiriac already find problem, thanks :+1:

File: src/parser/statement.js (hunk 2 above)
Comment (evilebottnawi) [reply src/parser/statement.js]: @ichiriac fixed already :+1:

File: src/parser/statement.js [hunk 3]
```diff
@@ -51,8 +51,13 @@ module.exports = {
         return this.read_trait();
       case this.tok.T_USE:
         return this.read_use_statement();
-      case this.tok.T_CONST:
-        return this.next().read_const_list();
+      case this.tok.T_CONST: {
+        const result = this.node("constantstatement");
+        const items = this.next().read_const_list();
+        this.expectEndOfStatement();
+        console.log(result);
```
Comment (ichiriac) [inline src/parser/statement.js]: oups, bad commit ? :smile:

Comment (evilebottnawi) [reply src/parser/statement.js]: @ichiriac yep, WIP on this :+1:

**Pull request merged**